internal/lsp/server.go

```diff
@@ -1510,7 +1510,7 @@ func (l *LanguageServer) handleTextDocumentDefinition(params types.DefinitionPar
 
 	query := oracle.DefinitionQuery{
 		// The value of Filename is used if the defn in the current buffer.
-		Filename: params.TextDocument.URI,
+		Filename: uri.ToRelativePath(l.client.Identifier, params.TextDocument.URI, l.workspaceRootURI),
```

Comment:
There's also
```go
func (l *LanguageServer) toRelativePath(fileURI string) string {
	return strings.TrimPrefix(l.toPath(fileURI), l.workspacePath()+string(os.PathSeparator))
}
```
Which could be updated to use your new function, and then we can use that here instead